### PR TITLE
Reduced logger level for #1773

### PR DESCRIPTION
**Pull request by adrian-kong**

### CODE REVIEW

**Comment by adrian-kong** [comment]
hello i am back for more pull requests #1773

**Comment by adrian-kong** [comment]
i was also thinking, what if we make it like

```java
public static <T> T sysctl(String name, T def){
        switch(def.getClass) ...
        // integer
        // string
        // etc...
}
```
would that be nicer?

File: oshi-core/src/main/java/oshi/util/platform/mac/SysctlUtil.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ public static int sysctl(String name, int def) {
         size_t.ByReference size = new size_t.ByReference(com.sun.jna.platform.mac.SystemB.INT_SIZE);
         Pointer p = new Memory(size.longValue());
         if (0 != SystemB.INSTANCE.sysctlbyname(name, p, size, null, size_t.ZERO)) {
```
[the remark this answers is not in the document]
Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/util/platform/mac/SysctlUtil.java:63]: @sonatype-lift ignore

File: oshi-core/src/main/java/oshi/util/platform/unix/freebsd/BsdSysctlUtil.java [hunk 2]
```diff
@@ -61,7 +61,7 @@ public static int sysctl(String name, int def) {
         size_t.ByReference size = new size_t.ByReference(new size_t(FreeBsdLibc.INT_SIZE));
         Pointer p = new Memory(size.longValue());
         if (0 != FreeBsdLibc.INSTANCE.sysctlbyname(name, p, size, null, size_t.ZERO)) {
```
[the remark this answers is not in the document]
Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/util/platform/unix/freebsd/BsdSysctlUtil.java:63]: @sonatype-lift ignore

File: oshi-core/src/main/java/oshi/util/platform/unix/openbsd/OpenBsdSysctlUtil.java [hunk 3]
```diff
@@ -65,7 +65,7 @@ public static int sysctl(int[] name, int def) {
         size_t.ByReference size = new size_t.ByReference(new size_t(OpenBsdLibc.INT_SIZE));
         Pointer p = new Memory(size.longValue());
         if (0 != OpenBsdLibc.INSTANCE.sysctl(name, name.length, p, size, null, size_t.ZERO)) {
```
[the remark this answers is not in the document]
Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/util/platform/unix/openbsd/OpenBsdSysctlUtil.java:67]: @sonatype-lift ignore

**Review by dbwiddis — APPROVED** [review]

**Comment by dbwiddis** [comment]
> i was also thinking, what if we make it like
> 
> ```java
> public static <T> T sysctl(String name, T def){
>         switch(def.getClass) ...
>         // integer
>         // string
>         // etc...
> }
> ```
> 
> would that be nicer?

It would.  If you're refacctoring, I'd really like to just have one Sysctl class but load the appropriate C library based on which BSD we're on (mac, Free, Open).

**Comment by adrian-kong** [comment]
> > i was also thinking, what if we make it like
> > ```java
> > public static <T> T sysctl(String name, T def){
> >         switch(def.getClass) ...
> >         // integer
> >         // string
> >         // etc...
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > would that be nicer?
> 
> It would. If you're refacctoring, I'd really like to just have one Sysctl class but load the appropriate C library based on which BSD we're on (mac, Free, Open).

i'll give it a go, just have to wrap my head around what the possible parameters are cus sometimes it doesn't return what the def type is and sometimes name is int[]

but i think we can merge this PR 
i'll open another PR with what i come up with (without linking anything yet maybe just the general structure so nothing breaks)

**Pull request merged**